### ISSUE TITLE
Use of $id in $ref is not working

### ISSUE THREAD

**Issue by sunilgunisetty** [everit-org/json-schema#321]
I have the following schema in 2 different files.

```
// ProductDescription.json
{
    "$schema" : "http://json-schema.org/draft-07/schema",
    "$id": "https://some-host.com/ProductDescription.json",
    "type": "object",
    "properties": {.
                   "uuid": {
                       "description": "The internal UUID uniquely identifying this entity",
                       "allOf": [
                           { "$ref": "/Types.json#uuid" }
                        ]
                    },    
       }
}
```
and
 
```
//Types.json
{
    "$schema" : "http://json-schema.org/draft-07/schema",
    "$id": "https://some-host.com/types.json",
    "definitions": {
        "uuid": {
            "$id": "#uuid",
            "type": "string",
            "pattern": "^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$"
        }
    }
}
```

When I try to load `ProductDescription.json` I get exception with a message `a JSON pointer should start with '/' or '#/' `.

I am using static `load` method in `SchemaLoader` class to load the schema.

In my front end I use [`ajv`](https://github.com/epoberezkin/ajv) with the same files and it works without any issue.

Am I missing something / am I using wrong methods. Any suggestions is highly appreciated.

**Comment by erosb** [comment]
Hello @sunilgunisetty , for now I suggest using the ` { "$ref": "/Types.json#/definitions/uuid" }` syntax, ie. referring to the subschema via a json pointer instead of its id.

I re-read the current version of the spec and I _think_ what you ask for _should_ work (though previous json schema drafts explained it differently). But it isn't covered by the official test suite, so I'm not 100% sure. Let's keep this issue open as a feature request.

**Comment by sunilgunisetty** [comment]
@erosb Thank you.

**Comment by eraserhd** [comment]
I think this says it should work? https://json-schema.org/latest/json-schema-core.html#rfc.section.8.2.4

**Comment by erosb** [comment]
This is now fixed and released in version 1.12.0.
